### PR TITLE
Add support to search for examinants

### PR DESCRIPTION
**Pull request by sycoso**
- typeahead works (but there are duplicates etc [e.g. 'Bless' and 'Bless
  et al' or 'Abeck' and 'Abeck; Hillebrand', blame the database :)])
- deleted dead code: function filtered() in TypeaheadList (was
  LectureList)

### CODE REVIEW

**Comment by sycoso** [comment]
Fixes #34

**Comment by sycoso** [comment]
I think the variable names are better now.

File: js/vm/typeaheadlist.js [hunk 1]
```diff
@@ -26,20 +26,11 @@ export default class LectureList {
...
-  filtered() {
-    if (this.searchString === '') {
-      return this;
-    }
-    let regex = this.getSearchRegex(this.searchString);
-
-    return this.lectures.filter(l => regex.test(l.name));
-  }
-
   get typeaheadDataset() {
```
Comment (Kha) [inline js/vm/typeaheadlist.js]: Now that the dataset isn't actually used anymore except for its `source` method, I think you can just delete it and inline the `source` implementation at the call site in `DocumentSelection`.

**Comment by Kha** [comment]
Excellent, this looks very nice!

File: js/vm/typeaheadlist.js (hunk 1 above)
Comment (sycoso) [inline js/vm/typeaheadlist.js]: There still is the ```getSearchRegex``` function. Should I put that in ```DocumentSelection``` too?

Comment (Kha) [inline js/vm/typeaheadlist.js]: Hm, I guess so. Of course, then we can ditch the whole `TypeaheadList` class and put the `getJSON` calls in the `DocumentSelection` constructor, which sounds even better.

**Comment by Kha** [comment]
Ha, I see you're one step ahead of me.

**Comment by sycoso** [comment]
I was bored :)

**Pull request merged**